### PR TITLE
resolves #3 - Implement latest mock methods for language manager

### LINKED ISSUE #3: MockLanguageManager does not support all available methods
A few methods have been added since AEM 6.5.6 which are very handy for certain situations (for example when not working with pages but with experience fragments or content fragments). This issue is to track the support of those methods

### PR DESCRIPTION
**Pull request by royteeuwen**
Fixes #3

### CODE REVIEW

File: core/src/test/java/io/wcm/testing/mock/aem/MockLanguageManagerTest.java [hunk 1]
```diff
@@ -280,4 +362,10 @@ public void getAdjacentInfo() {
     Map<Locale, LanguageManager.Info> rootPath = languageManager.getAdjacentInfo(context.resourceResolver(), "/");
     assertNull(rootPath);
   }
+
+  public void testGetLanguageRootResource() {
```
Comment (stefanseifert) [inline core/src/test/java/io/wcm/testing/mock/aem/MockLanguageManagerTest.java:366]: two empty methods? obsolete or do you wanted to add additional tests?

**Review by stefanseifert — CHANGES_REQUESTED** [review]
lgtm in general (although the methods tend to get a bit too big for a single method)

please also add an entry fior the changelog

File: core/src/test/java/io/wcm/testing/mock/aem/MockLanguageManagerTest.java (hunk 1 above)
Comment (royteeuwen) [reply core/src/test/java/io/wcm/testing/mock/aem/MockLanguageManagerTest.java]: nop auto-generated. Are incorporated in the other tests

**Comment by royteeuwen** [comment]
Updated changelog

**Review by stefanseifert — APPROVED** [review]

**Comment by stefanseifert** [comment]
@royteeuwen  we try to comply with sonarcloud checks for new code - quality gate failes by a small margin (code smells, coverage) - can you have another look? thanks!
https://sonarcloud.io/summary/new_code?id=wcm-io_io.wcm.testing.aem-mock&branch=feature%2Fadd-new-language-manager-mock-methods

**Comment by royteeuwen** [comment]
let's see what happens now :)! hopefully it will be enough now

**Comment by stefanseifert** [comment]
https://sonarcloud.io/summary/new_code?id=wcm-io_io.wcm.testing.aem-mock&branch=feature%2Fadd-new-language-manager-mock-methods
code coverage 85%
still 3 code smells concerning three overly complex methods - let's accept this for this mock (but usually it's worth and will produce better code to split the methods up smaller parts)

**Pull request merged**